### ISSUE TITLE
Unresolved relationship tables giving avoidable binding errors

### ISSUE THREAD

**Issue by semihsalihoglu-uw** [kuzudb/kuzu#2812]
I have a database with an RDFGraph TurtleTest, a regular node table Student, and a regular rel table st_r from student to TurtleTest_r resource node table. I get the following error:
```
kuzu> match (a)-[e]->(b:TurtleTest_l) return *;
Error: Binder exception: Relationship pattern e contains both property graph relationship label st_to_r and RDFGraph label TurtleTest. Mixing two tables in the same relationship pattern is not supported.
kuzu> call show_tables() return *;
```
I should not be getting this because the destination of e is TurtleTest_l, so e can only bind to TurtleTest_lt. So there is no real mixing of relationship tables from RDFGraph and regular property graph. We should not disable all multilabel queries because a database has an rdfgraph and a regular relationship table. We should only disable it when a relationship pattern resolves to mix these two types of relationship tables.

Here is a more basic example that we re disabling unlabeled or multilabel queries completely:
I have a database with an RDFGraph TestTurtle, a regular node table student, and a regular rel table st_to_st from student to student.
```
kuzu> create rel table st_to_st (from student to student);
-----------------------------------------
| result                                |
-----------------------------------------
| Rel table: st_to_st has been created. |
-----------------------------------------
 match (a)-[e]->(b:TurtleTest_l) return *;
Error: Binder exception: Relationship pattern e contains both property graph relationship label st_to_st and RDFGraph label TurtleTest. Mixing two tables in the same relationship pattern is not supported.
```
So as long as there is a regular relationship in the database we give this error even though there is no relationship table between an RDFGraph's node table and a regular node table.

**Comment by andyfengHKU** [comment]
> We should not disable all multilabel queries because a database has an rdfgraph and a regular relationship table

The fact user writes `(a)-[e]->(b:student)` meaning `e.iri` should appear in the return columns (even though it will be all NULL). So we need to compile an operator that writes `e.iri ValueVector` which will be an inner hash join based on our current setup and will lead to bug/incorrect result.

You seem to think `(a)-[e]->(b:student)` is the same as `(a:student)-[e:st_from_st]->(b:student)` but they are not. They may have similar (or even the same) computation but the return columns are different. We need to avoid the case where `e.iri` appears in the return column and can be NULL. 

You might then argue for `(a)-[e]->(b:TurtleTest_l)` we can indeed infer `e.iri` will never be NULL. This is true but it means we need to do checking after label pruning. I don't think we should write extra code to patch a limitation that we will remove soon (unless we don't want to support mixing RDFGraph label and PropertyGraph label in the near future).

To sum up, I'm not planning to solve this binding issue. If I were to solve, I would solve #2798 which automatically solves this.
